fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -817,7 +820,7 @@ public void createTable(CreateTableStmt stmt) throws DdlException {
             db.readUnlock();
         }
 
-        if (engineName.equalsIgnoreCase("olap")) {
+        if (stmt.isOlapOrLakeEngine()) {
```

Comment:
Does the name of  createOlapTable need to be changed?
BTW, is lake engine needs so many properties like (colocate, dynamic partition) similar to olap table? There are too many engine type check in the code, it will make these code hard to maintain.

Reply:
Currently, there are two differences between lake table and olap table
1. Lake table needs to get storage group from StarMgr.
2. Tablet is different.

Properties like colocate, dynamic partition may be supported later.